crates/uv/tests/common/mod.rs

```diff
@@ -96,6 +96,31 @@ impl TestContext {
         .success()
         .stdout(version);
     }
+
+    pub fn filters(&self) -> std::io::Result<Vec<(String, String)>> {
+        let cache_dir = self.cache_dir.canonicalize()?;
+        let venv_dir = self.venv.canonicalize()?;
+        let temp_dir = self.temp_dir.canonicalize()?;
+
+        let mut filters = INSTA_FILTERS
+            .iter()
+            .map(|(pattern, replacement)| (pattern.to_string(), replacement.to_string()))
+            .collect::<Vec<_>>();
+        filters.push((
+            cache_dir.to_str().unwrap().to_string(),
+            "[CACHE DIR]".to_string(),
+        ));
+        filters.push((
+            venv_dir.to_str().unwrap().to_string(),
+            "[VENV DIR]".to_string(),
+        ));
+        filters.push((
+            temp_dir.to_str().unwrap().to_string(),
+            "[TEMP DIR]".to_string(),
+        ));
+
+        Ok(filters)
+    }
```

Comment:
I want to return `Vec<(&str, &str)>` here where the lifetime is of the `TestContext` but can't because `cache_dir.canonicalize()` is owned by the function. Suggestions?

Reply:
Do I need to store the canonicalized path on the struct too?

Reply:
The reason this is a problem is e.g.

https://github.com/astral-sh/uv/pull/1781/files#diff-10e2684d108814c5407ca429afe35ac9916b5577d17b45afc4e800f6582f9ce7R949-R954

Reply:
If you can store the canonicalized path on `TestContext`, then that would do the trick I think?

But also, `Vec<(String, String)>` seems fine to me unless there's a reason not to do that?

Reply:
OK, so the reason why you want this is that `uv_snapshot!` wants a `Vec<(&str, &str)>`, is that right? I'm not familiar with `uv_snapshot!`, but if you can change it to accept `Vec<(impl AsRef<str>, impl AsRef<str>)>`, then that should make it work for both `Vec<(&str, &str)>` and `Vec<(String, String)>`.

Reply:
Just going to add it to the struct